api/transport/request.go

```diff
@@ -62,6 +62,10 @@ type Request struct {
 
 	// Request payload.
 	Body io.Reader
+
+	// TransportHeaders are other transport headers that the transport may expose
+	// to the encodings.
+	TransportHeaders map[string]string
```

Comment:
Did you see the PR description?
> I added an arbitrary TransportHeaders map[string]string, but I'm not sure if that's the right approach.

We need to pass some arbitrary information between the encoding and transport that is not stored in a standard header. I'm convinced this isn't the right approach, but wanted a simple proposal so it was obvious why we needed this.